src/libstore/build/derivation-goal.cc

```diff
@@ -803,7 +804,7 @@ void replaceValidPath(const Path & storePath, const Path & tmpPath)
        tmpPath (the replacement), so we have to move it out of the
        way first.  We'd better not be interrupted here, because if
        we're repairing (say) Glibc, we end up with a broken system. */
-    Path oldPath = fmt("%1%.old-%2%-%3%", storePath, getpid(), rand());
+    Path oldPath = fmt("%1%.old-%2%-%3%", storePath, getpid(), Random{}());
```

Comment:
I am not sure why in 68f4c728eca33f115f90e3f924c9081a4cd59896 I put it here, when it is only used in `LocalDerivationGoal`. Please move it to the right file, and then you can get the `Random` from the `LocalStore` that `LocalDerivationGoal` requires

Reply:
`void replaceValidPath(const Path & storePath, const Path & tmpPath)` is also just a free function. I am not sure how I'd access `rng` from `LocalStore`.